salt/states/git.py

```diff
@@ -1308,6 +1308,24 @@ def latest(name,
                         'if it does not already exist).',
                         comments
                     )
+                remote_tags = set([
+                    x.split('/')[-1] for x in __salt__['git.ls_remote'](
+                        cwd=target,
+                        remote=remote,
+                        opts="--tags",
+                        user=user,
+                        password=password,
+                        identity=identity,
+                        saltenv=__env__,
+                        ignore_retcode=True,
+                    ).keys() if '^{}' not in x
+                ])
+                if set(all_local_tags) != remote_tags:
+                    has_remote_rev = False
+                    ret['changes']['tags'] = {
+                        'old': all_local_tags,
+                        'new': list(remote_tags)
```

Comment:
Having a sub-dict with `new` probably isn't necessary. You could just do something like:

```python
ret['changes']['new tags'] = [x for x in remote_tags if x not in all_local_tags]
```

I haven't tested that code, so you'll want to double-check this and make sure it behaves correctly.